Website/Packages/datax-pipeline/src/modules/flowDefinition/flowModels.js

```diff
@@ -527,6 +645,44 @@ export function getMetricSinker() {
     };
 }
 
+export function getDefaultBatchSettings(type) {
+    if (type === batchTypeEnum.oneTime) {
+        return {
+            id: '',
+            type: type,
+            disabled: false,
+            properties: {
+                interval: '1',
+                intervalType: 'day',
+                delay: '',
+                delayType: '',
+                window: '1',
+                windowType: 'day',
+                startTime: '',
+                endTime: '',
+                lastProcessedTime: ''
+            }
+        };
+    } else {
+        return {
+            id: '',
+            type: type,
+            disabled: false,
+            properties: {
+                interval: '1',
+                intervalType: 'day',
+                delay: '0',
+                delayType: 'day',
+                window: '1',
```

Comment:
out of curiosity, why do we store all the numbers as strings? is this a backend limitations? otherwise, I would have expected the number fields being just a number as opposed to a string.

Reply:
no special reason. but I think we always handle the numbers this way. e.g. windowDuration and maxRate